coderd/autobuild/lifecycle_executor.go

```diff
@@ -269,15 +268,14 @@ func isEligibleForAutostop(build database.WorkspaceBuild, job database.Provision
 
 // isEligibleForFailedStop returns true if the workspace is eligible to be stopped
 // due to a failed build.
-func isEligibleForFailedStop(job database.ProvisionerJob, templateSchedule schedule.TemplateScheduleOptions) bool {
+func isEligibleForFailedStop(build database.WorkspaceBuild, job database.ProvisionerJob, templateSchedule schedule.TemplateScheduleOptions) bool {
 	// If the template has specified a failure TLL.
 	return templateSchedule.FailureTTL > 0 &&
 		// And the job resulted in failure.
 		db2sdk.ProvisionerJobStatus(job) == codersdk.ProvisionerJobFailed &&
+		build.Transition == database.WorkspaceTransitionStart &&
 		// And sufficient time has elapsed since the job has completed.
 		(job.CompletedAt.Valid && database.Now().Sub(job.CompletedAt.Time) > templateSchedule.FailureTTL ||
 			// Or sufficient time has elapsed since the job was canceled.
-			job.CanceledAt.Valid && database.Now().Sub(job.CanceledAt.Time) > templateSchedule.FailureTTL ||
-			// Or the job is stuck/abandoned.
-			database.Now().Sub(job.UpdatedAt) > 30*time.Second)
+			job.CanceledAt.Valid && database.Now().Sub(job.CanceledAt.Time) > templateSchedule.FailureTTL)
```

Comment:
Canceled jobs still get marked completed and marked failed so this might not be necessary. Could also indicate a hung/in-progress job if it's canceled but not completed, which you don't want to mess with